File: java/src/main/java/ps_traning/programmers/level_2/귤_고르기.java
package ps_traning.programmers.level_2;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class 귤_고르기 {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for (int size : tangerine) {
            map.put(size, map.getOrDefault(size, 0) + 1);
        }
        List<Integer> cnts = new ArrayList(map.values());
//        Collections.sort(cnts, Collections.reverseOrder());
        cnts.sort((n1, n2) -> n2 - n1);
        for (int cnt : cnts) {
            k -= cnt;
            answer++;
            if (k <= 0) break;
        }
        return answer;
    }
}

File: python/with_java/programmers/level 2/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    d = {}
    for t in tangerine:
        d[t] = d.get(t, 0) + 1
    cnts = sorted(list(d.values()), reverse=True)
    for cnt in cnts:
        k -= cnt
        answer += 1
        if k <= 0:
            break
    return answer